src/main/java/org/freeswitch/esl/client/internal/AbstractEslClientHandler.java

```diff
@@ -225,23 +226,51 @@ protected void handleEslMessage(ChannelHandlerContext ctx, EslMessage message) {
 	 * @param command
 	 * @return Job-UUID as a string
 	 */
-	public CompletableFuture<EslEvent> sendBackgroundApiCommand(Channel channel, final String command) {
-
+	public CompletableFuture<String> sendBackgroundApiJobCommand(Channel channel, final String command) {

```

Comment:
If i understand correctly, channelRead0 which is used to listen to command response when completed used the jobId to fetch future. If we only use sendBackgroundApiJobCommand method directly without populating backgroupJobs. Will the response be read. In other words, how do we want to use this method.

Reply:
This was the flow for the function `sendBackgroundApiCommand` 

`bgapi ---Sync---> JOB_UUID ----Async---> Job Events`

So we want the first to handle of `JOB_UUID` and not care of `Job Events` when we call  `sendBackgroundApiJobCommand`. This will let us use `sendBackgroundApiJobCommand` success/failure to appropriately send back in http-response.

Reply:
Got it.